functions/Get-DbaAgentJobHistory.ps1

```diff
@@ -0,0 +1,168 @@
+
+			Returns the SQL Agent Job execution results between 2017/05/22 00:00:00 and 2017/05/23 12:30:00 on sql2\Inst2K17.
+	
+		.EXAMPLE 
+	 		Get-DbaAgentJob -SqlInstance sql2016 | Where Name -match backup | Get-DbaAgentJobHistory
+	
+			Gets all jobs with the name that match the regex pattern "backup" and then gets the job history from those. You can also use -Like *backup* in this example.
+	
+	#>
+	[CmdletBinding(DefaultParameterSetName = "Default")]
+	param (
+		[parameter(Mandatory, ValueFromPipeline, ParameterSetName = "Server")]
+		[Alias("ServerInstance", "SqlServer")]
+		[DbaInstanceParameter[]]$SqlInstance,
+		[PSCredential][System.Management.Automation.CredentialAttribute()]
+		$SqlCredential,
+		[string[]]$Job,
+		[DateTime]$StartDate = "1900-01-01",
+		[DateTime]$EndDate = $(Get-Date),
+		[Switch]$NoJobSteps,
+		[parameter(Mandatory, ValueFromPipeline, ParameterSetName = "Collection")]
+		[Microsoft.SqlServer.Management.Smo.Agent.Job]$JobCollection,
+		[switch]$Silent
+	)
+	
+	begin {
+		$filter = New-Object Microsoft.SqlServer.Management.Smo.Agent.JobHistoryFilter
+		$filter.StartRunDate = $StartDate
+		$filter.EndRunDate = $EndDate
+		
+		function Get-JobHistory {
+			[CmdletBinding()]
+			param (
+				$Server,
+				$Job
+			)
+			
+			try {
+				Write-Message -Message "Attempting to get job history from $instance" -Level Verbose
+				if ($Job) {
+					foreach ($currentjob in $job) {
+						$filter.JobName = $currentjob
+						$executions += $server.JobServer.EnumJobHistory($filter)
+					}
+				}
+				else {
+					$executions = $server.JobServer.EnumJobHistory($filter)
+				}
+				
+				if ($NoJobSteps) {
+					$executions = $executions | Where-Object { $_.StepID -eq 0 }
+				}
+				
+				foreach ($execution in $executions) {
+					Add-Member -InputObject $execution -MemberType NoteProperty -Name ComputerName -value $server.NetName
+					Add-Member -InputObject $execution -MemberType NoteProperty -Name InstanceName -value $server.ServiceName
+					Add-Member -InputObject $execution -MemberType NoteProperty -Name SqlInstance -value $server.DomainInstanceName
+					
+					Select-DefaultView -InputObject $execution -Property ComputerName, InstanceName, SqlInstance, 'JobName as Job', StepName, RunDate, RunDuration, RunStatus
+				}
+			}
+			catch {
+				Stop-Function -Message "Could not get Agent Job History from $instance" -Target $instance -Continue
+			}
+		}
+	}
+	
+	process {
+		
+		if ($JobCollection) {
+			foreach ($currentjob in $JobCollection) {
+				Get-JobHistory -Server $currentjob.Parent.Parent -Job $currentjob.Name
+			}
+		}
+		
+		foreach ($instance in $SqlInstance) {
+			Write-Message -Message "Attempting to connect to $instance" -Level Verbose
+			try {
+				$server = Connect-SqlInstance -SqlInstance $instance -SqlCredential $SqlCredential
+			}
+			catch {
+				Stop-Function -Message "Could not connect to Sql Server instance $instance" -Target $instance -Continue
```

Comment:
We need to start including the InnerErrorRecord with these. I've come across some that have like access denied or something, it will help folks see why it didn't connect. Otherwise, you have to go find the exception file and look at that or do the extra dirty of typing out `$error[0]`